tests/e2e/bcd_consumer_integration_test.go

```diff
@@ -237,11 +239,54 @@ func (s *BCDConsumerIntegrationTestSuite) Test03BTCHeaderPropagation() {
 	s.Require().Equal(header1.Hash.MarshalHex(), consumerBtcHeaders.Headers[1].Hash)
 }
 
-// Test04CreateConsumerFinalityProvider
+// TestBTCTimestampRelay verifies that BTC timestamps are properly relayed from Babylon to the consumer chain
+func (s *BCDConsumerIntegrationTestSuite) Test04BTCTimestampRelay() {
+	// Get current Babylon epoch information
+	bbnClient := s.babylonController.GetBBNClient()
+	currentEpochResp, err := bbnClient.CurrentEpoch()
+	s.Require().NoError(err)
+	currentEpoch := currentEpochResp.CurrentEpoch
+
+	s.T().Logf("Current epoch: %d", currentEpoch)
+
+	// Wait for the current epoch to be sealed
+	var lastSealedCkpt *ckpttypes.QueryLastCheckpointWithStatusResponse
+	s.Eventually(func() bool {
+		lastSealedCkpt, err = bbnClient.LatestEpochFromStatus(ckpttypes.Sealed)
+		if err != nil {
+			return false
+		}
+		return currentEpoch <= lastSealedCkpt.RawCheckpoint.EpochNum
+	}, 1*time.Minute, 1*time.Second)
+
+	s.T().Logf("Epoch %d is now sealed", currentEpoch)
+
+	// Finalize the epoch
+	s.finalizeUntilEpoch(currentEpoch, true)
+
+	s.T().Logf("Epoch %d is now finalized", currentEpoch)
+
+	// Verify all BTC timestamps were relayed to the consumer chain
+	var consumerEpoch *cosmwasm.BabylonEpochResponse
+	s.Eventually(func() bool {
+		consumerEpoch, err = s.cosmwasmController.QueryBabylonLastEpoch()
+		return err == nil && consumerEpoch != nil &&
+			consumerEpoch.EpochNumber == currentEpoch
+	}, 1*time.Minute, 1*time.Second)
+
+	// Assert the BTC timestamp info is consistent between Babylon and consumer chain
+	for i := uint64(1); i <= currentEpoch; i++ {
+		s.validateEpochInfo(i, bbnClient)
+		s.validateCheckpointInfo(i, bbnClient)
+		s.validateConsumerHeader(i, bbnClient, consumerID)
+	}
```

Comment:
In fact, instead of having a new test case, wdyt we just add these assertions to the `finalizeUntilEpoch` function, so that we have implicitly tested the BTC timestamping integration?